packages/esm-form-render-app/src/render/forms-render-test.component.tsx

```diff
@@ -74,6 +75,65 @@ function FormRenderTest() {
     setIsSchemaLoaded(false);
   };
 
+  const formValidator = () => {
+    if (defaultJson) {
+      const parsedForm = typeof defaultJson == 'string' ? JSON.parse(defaultJson) : defaultJson;
+
+      for (let i = 0; i < parsedForm.pages.length; i++) {
+        for (let j = 0; j < parsedForm.pages[i].sections.length; j++) {
+          for (let k = 0; k < parsedForm.pages[i].sections[j].questions.length; k++) {
+            
+            const questionObject = parsedForm.pages[i].sections[j].questions[k];
+            handleFormValidation(questionObject);
+          }
+        }
+      }
+    } else {
+      console.log('Empty form!');
+    }
+  };
+
+  const handleFormValidation = (conceptObject) => {
+    conceptObject.questionOptions.concept
+      ? openmrsFetch(`/ws/rest/v1/concept/${conceptObject.questionOptions.concept}`)
+          .then((response) => {
+            console.log(response.data);
+            dataTypeChecker(conceptObject, response);
+          })
+          .catch((error) => {
+            error.message.includes('404') &&
+              console.log(`❌ Concept UUID ${conceptObject.questionOptions.concept} not found`);
+          })
+      : console.log('❌ Question with no concept UUID: ', conceptObject.id);
+  };
+
+  const dataTypeChecker = (conceptObject, responseObject) => {
+    const dataTypes = ['Numeric', 'Coded', 'Text', 'Date', 'Datetime', 'Boolean', 'Rule'];
+
+    const renderTypes = [
+      ['number'],
+      ['select', 'checkbox', 'radio', 'toggle', 'content-switcher'],
+      ['text', 'textarea'],
+      ['date'],
+      ['datetime'],
+      ['toggle', 'select', 'radio', 'content-switcher'],
+      ['repeating', 'group']
+    ];
+
+    if (conceptObject.questionOptions.concept === responseObject.data.uuid) {
+
+      dataTypes.forEach((dataType, index) => {
```

Comment:
Yeah, this actually will work better, thank you. Will implement this and give you feedback.

Reply:
This is done, although might be a little more expensive since it goes through every element in the renderTypes array. Kindly advice on the way forward.

Reply:
Actually it should be just one object and not a list. Something like:

```
dataTypes = {
'Coded':['select', 'checkbox', 'radio', 'toggle', 'content-switcher']
}
```
In this case you don't need a loop, you just do:

`dataTypes(responseObject.data.datatype.display)` or `dataTypes(responseObject.data.datatype.display) ` not sure what is the right syntax.

Reply:
Let me try this approach